File: typescript/8kyu/count_sheep/count_sheep.ts
export function countSheep(num: number): string {
    let result = "";
    for (let i = 1; i <= num; i++) {
        result += `${i} sheep...`;
    }
    return result;
}

// * Clever
// export function countSheep(num: number): string {
//     return Array.from({ length: num }, (_, i) => `${i + 1} sheep...`).join('');
// }

import { assert } from "chai";

describe("Sample Test Cases", function () {
    it("Should return a string", function () {
        assert.equal(countSheep(0), '');
        assert.equal(countSheep(1), '1 sheep...');
        assert.equal(countSheep(2), '1 sheep...2 sheep...');
        assert.equal(countSheep(3), '1 sheep...2 sheep...3 sheep...');
    });
});

File: dart/8kyu/count_sheep/bin/count_sheep.dart
import "package:test/test.dart";

String countSheep(int numb) {
  var result = '';
  for (var i = 1; i <= numb; i++) {
    result += '$i sheep...';
  }
  return result;
}

// * Clever
// String countSheep(numb) =>
//     List.generate(numb, (int i) => "${i + 1} sheep...").join();

void main() {
  group("Fixed tests", () {
    test("Testing for 0", () => expect(countSheep(0), equals("")));
    test("Testing for 1", () => expect(countSheep(1), equals("1 sheep...")));
    test("Testing for 2",
        () => expect(countSheep(2), equals("1 sheep...2 sheep...")));
    test("Testing for 3",
        () => expect(countSheep(3), equals("1 sheep...2 sheep...3 sheep...")));
  });
}
